app/src/main/java/app/pwhs/blockads/service/GoTunnelAdapter.kt

```
@@ -497,4 +510,77 @@ class GoTunnelAdapter(
+                DatagramPacket(
+                    query,
+                    query.size,
+                    InetAddress.getByName(host),
+                    port
+                )
+            )
+            val response = ByteArray(512)
+            val packet = DatagramPacket(response, response.size)
+            socket.receive(packet)
+            isValidHealthResponse(response, packet.length)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)